wandb/sdk/wandb_run.py

```diff
@@ -1469,47 +1469,68 @@ def _datatypes_callback(self, fname: str) -> None:
         files: FilesDict = dict(files=[(GlobStr(glob.escape(fname)), "now")])
         self._backend.interface.publish_files(files)
 
-    def _visualization_hack(self, row: dict[str, Any]) -> dict[str, Any]:
+    def _visualization_hack(
+        self,
+        data: dict[str, Any],
+    ) -> dict[str, Any]:
+        """Transforms visualization objects into table objects for a given dictionary of data.
+
+        This function handles special visualization types (Visualize and CustomChart) by:
+        - For Visualize objects: Replaces them with their underlying table
+        - For CustomChart objects:
+           - Replaces the CustomChart object with its table in the log history
+           - Modifies the config to use the table key when adding the graphgraphing
```

Comment:
this is too tide to the current specific implementation, maybe explain the general idea of this helper function